### PR TITLE
GH-181 - code fix provider for NS5001

### LINKED ISSUE #181: Add code fix for NS5001
Add code fix provider which removes redundant `Received` call from `Received.InOrder` checks

Before fix
````c#
using NSubstitute;
using NSubstitute.ReceivedExtensions;

namespace MyNamespace
{
    public interface IFoo
    {
        int Bar();
        int Bar(int x);
    }

    public class FooTests
    {
        public void Test()
        {
            var substitute = NSubstitute.Substitute.For<IFoo>();
            Received.InOrder(() =>
            { 
                substitute.Received().Bar();
                substitute.Received().Bar(Arg.Any<int>());
            });
        }
    }
}
````
after fix
````c#
using NSubstitute;
using NSubstitute.ReceivedExtensions;

namespace MyNamespace
{
    public interface IFoo
    {
        int Bar();
        int Bar(int x);
    }

    public class FooTests
    {
        public void Test()
        {
            var substitute = NSubstitute.Substitute.For<IFoo>();
            Received.InOrder(() =>
            { 
                substitute.Bar();
                substitute.Bar(Arg.Any<int>());
            });
        }
    }
}
````

### PR DESCRIPTION
**Pull request by tpodolak**
Closes #181 
receivedcodefix

### CODE REVIEW

File: src/NSubstitute.Analyzers.Shared/CodeFixProviders/AbstractReceivedInReceivedInOrderCodeFixProvider.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DiagnosticIdentifiers.ReceivedUsedInReceivedInOrder);
+
+    public override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        var diagnostic = context.Diagnostics.FirstOrDefault(diag => diag.Descriptor.Id == DiagnosticIdentifiers.ReceivedUsedInReceivedInOrder);
+        if (diagnostic != null)
+        {
+            var codeAction = CodeAction.Create("Remove redundant checks", ct => CreateChangedDocument(ct, context, diagnostic), nameof(AbstractReceivedInReceivedInOrderCodeFixProvider));
```
Comment (tpodolak) [inline src/NSubstitute.Analyzers.Shared/CodeFixProviders/AbstractReceivedInReceivedInOrderCodeFixProvider.cs:24]: @dtchepak should we include check name in title? E.g "Remove redundant Received check", "Remove redundant ReceivedWithAnyArgs check" etc

Comment (dtchepak) [reply src/NSubstitute.Analyzers.Shared/CodeFixProviders/AbstractReceivedInReceivedInOrderCodeFixProvider.cs:24]: I think "Remove redundant Received checks" would be great. (I don't think we need to change it for `ReceivedWithAnyArgs`, they are all `Received*` family checks)

File: tests/NSubstitute.Analyzers.Tests.CSharp/CodeFixProviderTests/ReceivedInReceivedInOrderCodeFixProviderTests/ReceivedAsExtensionMethodTests.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    public class FooTests
+    {
+        public void Test()
+        {
+            var substitute = NSubstitute.Substitute.For<IFoo>();
+            Received.InOrder(() =>
+            { 
+                substitute.Received().Bar();
+                substitute.Received().Bar(Arg.Any<int>());
```
Comment (dtchepak) [inline tests/NSubstitute.Analyzers.Tests.CSharp/CodeFixProviderTests/ReceivedInReceivedInOrderCodeFixProviderTests/ReceivedAsExtensionMethodTests.cs:30]: **suggestion:** maybe add a `substitute.ReceivedWithAnyArgs().Bar(Arg.Any<int>()))` call to this test as well?

**Review by dtchepak — APPROVED** [review]

File: tests/NSubstitute.Analyzers.Tests.CSharp/CodeFixProviderTests/ReceivedInReceivedInOrderCodeFixProviderTests/ReceivedAsExtensionMethodTests.cs (hunk 2 above)
Comment (tpodolak) [reply tests/NSubstitute.Analyzers.Tests.CSharp/CodeFixProviderTests/ReceivedInReceivedInOrderCodeFixProviderTests/ReceivedAsExtensionMethodTests.cs:30]: added

File: src/NSubstitute.Analyzers.Shared/CodeFixProviders/AbstractReceivedInReceivedInOrderCodeFixProvider.cs (hunk 1 above)
Comment (tpodolak) [reply src/NSubstitute.Analyzers.Shared/CodeFixProviders/AbstractReceivedInReceivedInOrderCodeFixProvider.cs]: sounds good, changed

**Pull request merged**